File: src/Sample/Plumix.Sample/Demos/Cupertino/ContextMenuDemoPage.cs
using Avalonia;
using Avalonia.Media;
using Plumix.Cupertino;
using Plumix.Rendering;
using Plumix.Widgets;

namespace Plumix;

// Dart parity source (reference): dart_sample/lib/demos/cupertino/context_menu_demo_page.dart

public sealed class CupertinoContextMenuDemoPage : StatefulWidget
{
    public override State CreateState() => new CupertinoContextMenuDemoPageState();
}

internal sealed class CupertinoContextMenuDemoPageState : State
{
    private string _lastAction = "none";

    public override Widget Build(BuildContext context)
    {
        Widget preview = new Container(
            width: 180.0,
            height: 120.0,
            decoration: new BoxDecoration(
                Color: Color.FromUInt32(0xFF5E5CE6),
                BorderRadius: BorderRadius.Circular(16.0)),
            alignment: Alignment.Center,
            child: new Text(
                "Press and hold",
                color: Colors.White,
                fontSize: 18.0));

        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 18.0,
            children:
            [
                new Text("Cupertino context menu", fontSize: 20.0, color: Colors.Black),
                new Text(
                    "Hold the preview, choose an action, or drag the open preview down to dismiss.",
                    fontSize: 14.0,
                    color: Color.FromUInt32(0x8A000000)),
                new Center(
                    child: new CupertinoContextMenu(
                        actions:
                        [
                            BuildAction(context, "Copy", CupertinoIcons.DocOnDoc, "copy"),
                            BuildAction(context, "Share", CupertinoIcons.Share, "share"),
                            BuildAction(
                                context,
                                "Delete",
                                CupertinoIcons.Delete,
                                "delete",
                                destructive: true),
                        ],
                        child: preview,
                        enableHapticFeedback: true)),
                new Text(
                    $"last action: {_lastAction}",
                    fontSize: 13.0,
                    color: Color.FromUInt32(0xFF455A64)),
            ]);
    }

    private Widget BuildAction(
        BuildContext context,
        string label,
        IconData icon,
        string result,
        bool destructive = false)
    {
        return new CupertinoContextMenuAction(
            child: new Text(label),
            trailingIcon: icon,
            isDestructiveAction: destructive,
            onPressed: () =>
            {
                Navigator.Pop(context);
                SetState(() => _lastAction = result);
            });
    }
}

File: dart_sample/lib/demos/cupertino/context_menu_demo_page.dart
import 'package:cupertino_ui/cupertino_ui.dart';
import 'package:material_ui/material_ui.dart';

class CupertinoContextMenuDemoPage extends StatefulWidget {
  const CupertinoContextMenuDemoPage({super.key});

  @override
  State<CupertinoContextMenuDemoPage> createState() =>
      _CupertinoContextMenuDemoPageState();
}

class _CupertinoContextMenuDemoPageState
    extends State<CupertinoContextMenuDemoPage> {
  String _lastAction = 'none';

  @override
  Widget build(BuildContext context) {
    final Widget preview = Container(
      width: 180,
      height: 120,
      decoration: const BoxDecoration(
        color: Color(0xFF5E5CE6),
        borderRadius: BorderRadius.all(Radius.circular(16)),
      ),
      alignment: Alignment.center,
      child: const Text(
        'Press and hold',
        style: TextStyle(color: Colors.white, fontSize: 18),
      ),
    );

    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 18,
      children: <Widget>[
        const Text(
          'Cupertino context menu',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'Hold the preview, choose an action, or drag the open preview '
          'down to dismiss.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Center(
          child: CupertinoContextMenu(
            enableHapticFeedback: true,
            actions: <Widget>[
              _buildAction(context, 'Copy', CupertinoIcons.doc_on_doc, 'copy'),
              _buildAction(context, 'Share', CupertinoIcons.share, 'share'),
              _buildAction(
                context,
                'Delete',
                CupertinoIcons.delete,
                'delete',
                destructive: true,
              ),
            ],
            child: preview,
          ),
        ),
        Text(
          'last action: $_lastAction',
          style: const TextStyle(fontSize: 13, color: Color(0xFF455A64)),
        ),
      ],
    );
  }

  Widget _buildAction(
    BuildContext context,
    String label,
    IconData icon,
    String result, {
    bool destructive = false,
  }) {
    return CupertinoContextMenuAction(
      trailingIcon: icon,
      isDestructiveAction: destructive,
      onPressed: () {
        Navigator.pop(context);
        setState(() => _lastAction = result);
      },
      child: Text(label),
    );
  }
}
